### ISSUE TITLE
Support for tests connecting to stripe-mock via a Unix socket

### ISSUE THREAD

**Issue by seifertm** [stripe/stripe-python#486]
`stripe-mock` has the ability to listen on a Unix socket for incoming requests.
Currently, the tests of `stripe-python` only support connection via http, as far as I can tell. It would be nice, if we could configure the tests to connect to stripe-mock via a unix socket. Is this possible?

Background:
I am currently trying to update the package for the Stripe client for Gentoo Linux (see https://github.com/gentoo/gentoo/pull/10156). Gentoo packages should generally have a test phase that run some tests to validate their correct installation. We are trying to use `stripe-mock` in the test phase of `stripe-python`. However, it is currently necessary to determine a free port that `stripe-mock` can bind to, because we don't know which ports are used on the system where the `stripe-python` is being installed installed.

**Comment by ob-stripe** [comment]
Hey @seifertm. stripe-python has a few different HTTP clients depending on available libraries but I don't think any of them support UNIX sockets out of the box.

I haven't tried it, but you may be able to use [requests-unixsocket](https://github.com/msabramo/requests-unixsocket) for this. Install the package (with `pipenv` or `pip` or however you prefer), then set the API base and client like this:
```python
session = requests_unixsocket.Session()
client = stripe.http_client.RequestsClient(session=session)
stripe.default_http_client = client

stripe.api_base = 'http+unix://%2Fpath%2Fto%2Fsocket'
```

**Comment by seifertm** [comment]
Thank for for your quick response!

I decided to try to start stripe-mock on random ports and send a "health check" request via curl to see if it started.

I think if we patched `stripe-python` to use a non-official HTTP client and a different connection URL it created more maintenance overhead than the current solution.

I also understand that adding support for UNIX socket connections is probably not worth it if no supported client library supports it. I will therefore close this issue.

Thanks for your help anyway!
